client/src/main/java/org/eclipse/hono/client/AbstractConsumer.java

```diff
@@ -11,14 +11,13 @@
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
 
-package org.eclipse.hono.client.impl;
+package org.eclipse.hono.client;
```

Comment:
I wonder if it is necessary to move the abstract base classes from impl to the top level package. In the end they are implementation detail and are not referenced in any of the interfaces/types, right? So, if possible, IMHO we should keep them in the impl package. The concrete implementations are created using factory methods in the exported interfaces anyway ....

Reply:
Probably you are right - I hope that this does not cause visibility problems in an OSGi environment.
Will soon push an update where the classes are moved, if this does not work we can move them back anyway.